_posts/2021-07-05-pcp-grafana.md

```diff
@@ -0,0 +1,106 @@
+
+![Overview usage card](/images/overview-card-usage.png)
+
+That link leads you to the Metrics page, which in a minimal installation only shows current metrics:
+
+![Metrics page without PCP](/images/metrics-page-no-pcp.png)
+
+For spotting performance problems in the past, we need to enable historical metrics, which are provided by [Performance Co-Pilot](https://pcp.io/), aka.  "PCP".
+
+The "Install cockpit-pcp" button installs the necessary support for this, and asks you to log in again (the new cockpit-pcp bridge cannot be used in an existing session). After that, the Metrics page will start to show historical metrics. Combined with Cockpit's other pages such as "Services", "Logs", or "Podman containers", this is a very powerful tool to find performance related problems:
+
+<iframe width="960" height="720" src="https://www.youtube.com/embed/8_NLoJBjay4?rel=0" frameborder="0" allowfullscreen></iframe>
+
+
+## Metrics of multiple machines
+
+But what if you want to monitor not just one machine, but your whole fleet?
+
+Cockpit used to have a "Dashboard" to see some basic resource usage of up to twenty machines, but this was very inflexible, inefficient, insecure, lacking important features such as alerting, and was bound to a running Cockpit session. That's why the Dashboard got removed in [Cockpit 234](https://cockpit-project.org/blog/cockpit-234.html).
+
+One of today's industry standard solutions for visualizing metrics is [Grafana](https://grafana.com/); it does not have above limitations. PCP offers a [Grafana plugin](https://grafana-pcp.readthedocs.io/en/latest/) to provide the machine's metrics to a local or remote Grafana installation.
+
+## Setup of monitored machines
+
+Setting up the machines for sending metrics to Grafana involves a couple of steps:
+
+ * Install PCP and enable `pmlogger.service` to start data collection. This already happened with the "Install cockpit-pcp" button above.
+ * Enable `pmproxy.service`, the PCP component which exposes a read-only metrics query API to the network.
+ * pmproxy needs the [Redis database](https://redis.io/), so the `redis` package needs to be installed, and the service configured to start with `pmproxy.service`.
+ * pmproxy opens a network port (44322 by default), which needs to be opened in the firewall for the zone that contains the Grafana machine.
+
+The most recent Cockpit version can do all these steps for you with the "Metrics settings" button at the top of the page. That opens a dialog where you can enable "Export to network":
+
+![Metrics settings with enabled pmproxy](/images/metrics-settings-2.png)
+
+This will let you confirm any extra package that needs to be installed (such as redis). After that, you may get an alert for opening the corresponding firewall service:
```

Comment:
```suggestion
This will let you confirm any extra packages that need to be installed (such as redis). After that, you may get an alert for opening the corresponding firewall service:
```